src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -98,36 +104,66 @@ export const UserPropsProvider: FC = ({ children }) => {
     logout({ returnTo: returnUrl });
   };
 
+  const fetchUserProfile = useCallback(
+    async (impersonationData?: ImpersonationData) => {
+      const header = {
+        'tenant-key': `${tenantConfig.id}`,
+        'X-SESSION-ID': await getsessionId(),
+        Authorization: `Bearer ${token}`,
+        'x-locale': locale,
+      };
+      try {
+        const response = await fetch(
+          `${process.env.API_ENDPOINT}/app/profile`,
+          {
+            method: 'GET',
+            headers: setHeaderForImpersonation(header, impersonationData),
+          }
+        );
+
+        if (!response.ok) {
+          const errorData = await response.json();
+          throw new APIError(response.status, errorData);
+        }
+        return response.status === 204 ? true : await response.json();
+      } catch (err) {
+        console.error('Error during API call:', err);
+        throw err;
+      }
+    },
+    [token, tenantConfig.id, locale, getsessionId]
```

Comment:
Is getsessionId needed in the dependency array? It is an imported function and will not change.